pkg/v1/tkg/web/src/app/views/landing/azure-wizard/provider-step/azure-provider-step.component.ts

```diff
@@ -8,28 +8,53 @@ import { debounceTime, distinctUntilChanged, finalize, takeUntil } from 'rxjs/op
 
 import { APIClient } from '../../../../swagger/api-client.service';
 import { StepFormDirective } from '../../wizard/shared/step-form/step-form';
-import { TkgEventType } from '../../../../shared/service/Messenger';
+import { TkgEvent, TkgEventType } from '../../../../shared/service/Messenger';
 import { AzureResourceGroup } from './../../../../swagger/models/azure-resource-group.model';
 import { AzureWizardFormService } from 'src/app/shared/service/azure-wizard-form.service';
 import { ValidationService } from '../../wizard/shared/validation/validation.service';
 import Broker from 'src/app/shared/service/broker';
+import { FormMetaDataStore } from '../../wizard/shared/FormMetaDataStore';
+import { NotificationTypes } from '../../../../shared/components/alert-notification/alert-notification.component';
+
+enum ProviderField {
+    AZURECLOUD = 'azureCloud',
+    CLIENT = 'clientId',
+    CLIENTSECRET = 'clientSecret',
+    SUBSCRIPTION = 'subscriptionId',
+    TENANT = 'tenantId',
+    SSHPUBLICKEY = 'sshPublicKey',
+    REGION = 'region',
+    RESOURCEGROUPOPTION = 'resourceGroupOption',
+    RESOURCEGROUPEXISTING = 'resourceGroupExisting',
+    RESOURCEGROUPCUSTOM = 'resourceGroupCustom',
+}
+
+enum ResourceGroupOption {
```

Comment:
If you want to use these enums in the azure-provider-step.component.html you can create a getter method in the components class like:

 public get ResourceGroupOption() {
        return ResourceGroupOption;
    }
    
Then you can replace the strings such as 'existing' in the html with ResourceGroupOption.EXISTING